DevSkim-DotNet/Microsoft.DevSkim/Ruleset.cs

```diff
@@ -21,6 +24,116 @@ public class RuleSet : IEnumerable<Rule>
         public RuleSet()
         {
             _rules = new List<Rule>();
+            _oatRules = new List<ConvertedOatRule>();
+        }
+
+        private Regex searchInRegex = new Regex(".*\\((.*),(.*)\\)",RegexOptions.Compiled);
+
+
+        public ConvertedOatRule? DevSkimRuleToConvertedOatRule(Rule rule)
+        {
+            var clauses = new List<Clause>();
+            int clauseNumber = 0;
+            var expression = new StringBuilder("(");
+            foreach(var pattern in rule.Patterns ?? Array.Empty<SearchPattern>())
+            {
+                if (pattern.Pattern != null)
+                {
+                    clauses.Add(new ScopedRegexClause(pattern.Scopes ?? new PatternScope[] { PatternScope.All })
+                    {
+                        Label = clauseNumber.ToString(CultureInfo.InvariantCulture),
+                        Data = new List<string>() { pattern.Pattern },
+                        Capture = true,
+                        Arguments = pattern.Modifiers?.ToList() ?? new List<string>()
+                    });
+                    if (clauseNumber > 0)
+                    {
+                        expression.Append(" OR ");
+                    }
+                    expression.Append(clauseNumber);
+                    clauseNumber++;
+                }
+            }
+            if (clauses.Any())
+            {
+                expression.Append(")");
+            }
+            else
+            {
+                return new ConvertedOatRule(rule.Id,rule);
+            }
+            foreach(var condition in rule.Conditions ?? Array.Empty<SearchCondition>())
+            {
+                if (condition.Pattern?.Pattern != null)
```

Comment:
Just a nit but for more clarity is there a better way to name the Pattern field so the object and field are easier to distinguish rather than Pattern.Pattern.